app/src/androidTest/java/com/sdp/movemeet/Security/ProfileActivitySecurityTest.java

```diff
@@ -0,0 +1,29 @@
+package com.sdp.movemeet.Security;
+
+import androidx.test.ext.junit.rules.ActivityScenarioRule;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+
+import com.sdp.movemeet.ProfileActivity;
+import com.sdp.movemeet.R;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.assertion.ViewAssertions.matches;
+import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+
+@RunWith(AndroidJUnit4.class)
+
+public class ProfileActivitySecurityTest {
+
+    @Rule
+    public ActivityScenarioRule<ProfileActivity> testRule = new ActivityScenarioRule<>(ProfileActivity.class);
+
+    @Test
+    public void redirectionTest() {
+        onView(withId(R.id.button_login)).check(matches(isDisplayed()));
```

Comment:
So here, you test (as in the other tests) that the button button_login is not hidden, correct?

Reply:
Sort of. Actually, this test is here to test the redirection of unlogged user to login page. So in all these tests, I try to instantiate a class (for example in this case the profile page) without logging in. If the redirection works, we are supposed to be on the Login page and to test that, I look if the "login" button is displayed. If this is the case, it means that the security has worked.